wire-library/wire-runtime-swift/src/main/swift/wellknowntypes+Custom/Timestamp+Codable.swift

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 Block Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+extension TimeInterval {
+    func decomposed() -> (seconds: Int64, nanos: Int32) {
+        let seconds = Int64(self.rounded())
+        let remainder = self.remainder(dividingBy: 1)
+        let nanos = Int32(remainder * pow(10, 9))
+
+        return (seconds, nanos)
+    }
+
+    init(seconds: Int64, nanos: Int32) {
+        self.init(seconds)
+        self += TimeInterval(nanos) / pow(10, 9)
```

Comment:
nit: can we use a constant here for `pow(10, 9)`. For speeeeeed. L23 as well. Could be a private static constant like `nanosecondsPerSecond` too.